backend/uchicago/watcher.py

```diff
@@ -0,0 +1,67 @@
+import requests
+from lxml import html
+import pyrebase
+from difflib import Differ
+import sendgrid
+import os
+from sendgrid.helpers.mail import *
+
+FIREBASE = pyrebase.initialize_app({
+    'apiKey':
+    'AIzaSyCjBDyhwbXcp9kEIA2pMHLDGxmCM4Sn6Eg',
+    'authDomain':
+    'canigraduate-43286.firebaseapp.com',
+    'databaseURL':
+    'https://canigraduate-43286.firebaseio.com',
+    'storageBucket':
+    'canigraduate-43286.appspot.com',
+    'serviceAccount':
+    './service_account_key.json'
+})
+
+def watcher(program):
+	url = db.child('programs').child(program).child('metadata').child('catalog').get().val()
+	xpath = db.child('programs').child(program).child('metadata').child('source').get().val()
+	stored_requirement = db.child('programs').child(program).child('page-state').get().val()
+	tree = html.fromstring(requests.get(url).content)
+	current_requirement = tree.xpath(xpath)[0].text_content()
+	if stored_requirement == None:
+		db.child('programs').child(program).child('page-state').set(current_requirement)
+	elif stored_requirement != current_requirement:
+		db.child('programs').child(program).child('page-state').set(current_requirement)
+		emailHelper(program, stored_requirement, current_requirement)
+
+	extensions = db.child('programs').child(program).child('extensions').get().val()
+	if extensions != None:
```

Comment:
You can avoid the check by just doing `for e in (extensions or []):` on the next line.

Reply:
👍